### PR TITLE
[Cpp API Compatibility] Add at::broadcast_to compat interface

### PR DESCRIPTION
**Pull request by youge325**
## Summary
Add `at::broadcast_to` compat interface in Paddle C++ API compat layer.

## Changes
- `paddle/phi/api/include/compat/ATen/ops/broadcast_to.h`: New compat implementation that delegates to existing `expand()`
- `paddle/phi/api/include/compat/ATen/core/TensorBody.h`: Add `Tensor::broadcast_to()` method declaration
- `paddle/phi/api/include/compat/ATen/Functions.h`: Include new header

## Test Coverage
- Cross-framework test: `BroadcastToTest.cpp` with 13 test cases
- Shape coverage: scalar, small, large, multi-dim, same-shape, zero-dim
- Dtype coverage: kFloat, kDouble, kInt, kLong
- Edge cases: incompatible shape exception

## Verification
- Paddle build: pass
- ctest (ATen|c10|torch): 68/68 pass
- PCAT result_cmp: `paddle_BroadcastToTest` and `torch_BroadcastToTest` MATCH

### CODE REVIEW

File: paddle/phi/api/include/compat/ATen/ops/broadcast_to.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+namespace at {
+
+// broadcast_to - broadcasts a tensor to a new shape
+// In PyTorch, broadcast_to is functionally equivalent to expand,
+// with the key difference that broadcast_to does not share storage
+// with the original tensor (returns a view that may or may not share
+// storage depending on the implementation).
+//
+// Semantically: broadcast_to(self, size) == expand(self, size)
+inline Tensor broadcast_to(const Tensor& self, at::IntArrayRef size) {
```
Comment (risemeup1111) [inline paddle/phi/api/include/compat/ATen/ops/broadcast_to.h]: P1 **优先级：P1**

这里目前只补了 `IntArrayRef` 入口，但 PyTorch 的 `broadcast_to` 兼容面还包含 `SymInt` 入口：上游 schema 是 `broadcast_to(Tensor self, SymInt[] size)`，生成的 C++ API 也暴露 `at::broadcast_to_symint` / `Tensor::broadcast_to_symint`。当前 compat 里的 `c10::SymIntArrayRef` 不能隐式转成 `at::IntArrayRef`，所以迁移代码里如果调用 `at::broadcast_to_symint(t, shape)` 或 `t.broadcast_to_symint(shape)` 仍然会编译失败。

建议和现有 `reshape_symint`、`split_symint`、`full_symint` 等接口保持一致，补齐 `SymIntArrayRef` 的函数和成员入口，并加一个覆盖 `c10::SymIntArrayRef` 调用的 C++ 测试。实现形态可以类似：

```cpp
inline Tensor broadcast_to_symint(const Tensor& self,
                                  c10::SymIntArrayRef size) {
  std::vector<int64_t> size_vec;
  size_vec.reserve(size.size());
  for (const auto& dim : size) {
    size_vec.push_back(static_cast<int64_t>(dim));
  }
  return at::expand(self, at::IntArrayRef(size_vec));
}

inline Tensor Tensor::broadcast_to_symint(c10::SymIntArrayRef size) const {
  return at::broadcast_to_symint(*this, size);
}
```

同时在 `TensorBody.h` 中声明 `Tensor broadcast_to_symint(c10::SymIntArrayRef size) const;`。

**Review by risemeup1111 — CHANGES_REQUESTED** [review]
已完成首轮 review。兼容接口还有一个需要修复的阻塞问题，细节见行级评论。

- P3 **优先级：P3** 非行级：当前 `Check` 失败在 PR 模板校验，日志显示 `PR Category`、`PR Types` 和“是否引起精度变化”字段未按模板填写。建议按 develop 分支 PR 模板补齐这些字段后重新触发检查。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>
